libsolidity/formal/CHC.cpp

```diff
@@ -1733,40 +1809,35 @@ smtutil::Expression CHC::predicate(Predicate const& _block)
 	solAssert(false, "");
 }
 
-smtutil::Expression CHC::predicate(FunctionCall const& _funCall)
+smtutil::Expression CHC::predicate(
+	FunctionDefinition const* _funDef,
+	Expression const* _calledExpr,
+	FunctionType const* _funType,
+	std::vector<ASTPointer<Expression const>> _arguments,
+	smtutil::Expression _contractAddressValue
+)
 {
-	FunctionType const& funType = dynamic_cast<FunctionType const&>(*_funCall.expression().annotation().type);
-	auto kind = funType.kind();
+	solAssert(_calledExpr, "");
+	solAssert(_funType, "");
+	auto kind = _funType->kind();
 	solAssert(kind == FunctionType::Kind::Internal || kind == FunctionType::Kind::External || kind == FunctionType::Kind::BareStaticCall, "");
-
-	solAssert(m_currentContract, "");
-	auto function = functionCallToDefinition(_funCall, currentScopeContract(), m_currentContract);
-	if (!function)
+	if (!_funDef)
 		return smtutil::Expression(true);
 
-	auto contractAddressValue = [this](FunctionCall const& _f) {
-		auto [callExpr, callOptions] = functionCallExpression(_f);
-
-		FunctionType const& funType = dynamic_cast<FunctionType const&>(*callExpr->annotation().type);
-		if (funType.kind() == FunctionType::Kind::Internal)
-			return state().thisAddress();
-		if (MemberAccess const* callBase = dynamic_cast<MemberAccess const*>(callExpr))
-			return expr(callBase->expression());
-		solAssert(false, "Unreachable!");
-	};
 	errorFlag().increaseIndex();
-	std::vector<smtutil::Expression> args{errorFlag().currentValue(), contractAddressValue(_funCall), state().abi(), state().crypto(), state().tx(), state().state()};
 
-	auto const* contract = function->annotation().contract;
+	std::vector<smtutil::Expression> args{errorFlag().currentValue(), _contractAddressValue, state().abi(), state().crypto(), state().tx(), state().state()};
+
+	auto const* contract = _funDef->annotation().contract;
 	auto const& hierarchy = m_currentContract->annotation().linearizedBaseContracts;
-	solAssert(kind != FunctionType::Kind::Internal || function->isFree() || (contract && contract->isLibrary()) || util::contains(hierarchy, contract), "");
+	solAssert(kind != FunctionType::Kind::Internal || _funDef->isFree() || (contract && contract->isLibrary()) || util::contains(hierarchy, contract), "");
 
 	if (kind == FunctionType::Kind::Internal)
 		contract = m_currentContract;
 
 	args += currentStateVariables(*contract);
-	args += symbolicArguments(_funCall, contract);
-	if (!usesStaticCall(_funCall))
+	args += symbolicArguments(_funDef, _calledExpr, _funType, _arguments);
+	if (!((_funDef && (_funDef->stateMutability() == StateMutability::Pure || _funDef->stateMutability() == StateMutability::View)) || kind == FunctionType::Kind::BareStaticCall))
```

Comment:
I think it would make sense to add an overload for `usesStaticCall` for this case.
The function name is self-explanatory, but this condition really is not.